components/lsp-form/RedeemForm.tsx

```diff
@@ -30,13 +40,52 @@ const RedeemForm: FC<Props> = ({
   longTokenDecimals,
   shortTokenBalance,
   shortTokenDecimals,
+  lspContract,
+  erc20Contract,
+  address,
+  setCollateralBalance,
+  refetchLongTokenBalance,
+  refetchShortTokenBalance,
 }) => {
-  const [collateral, setCollateral] = useState("");
+  const [collateral, setCollateral] = useState("uma");
   const [amount, setAmount] = useState("");
   const [longTokenAmount, setLongTokenAmount] = useState("");
   const [shortTokenAmount, setShortTokenAmount] = useState("");
   const [collateralOnTop, setCollateralOnTop] = useState(false);
 
+  const redeem = useCallback(async () => {
+    if (lspContract && erc20Contract && longTokenAmount && shortTokenAmount) {
+      // Note: You need to have an equal amount of long and short tokens in order to redeem.
+      // So what value you use of long token or short token should be arbitrary, as long as both are equal.
+      // It should be noted too, as you can have more of one than the other, we need to make sure the user is not trying to redeem more tokens than they have of the matching pair.
+      const weiAmount = toWeiSafe(longTokenAmount);
+      try {
+        // Need to send the correct amount based on the collateral pair ** the amount
+        // User has specified in the input.
+        // All operations that make the number larger come first. All the operations that make the number smaller come last.
+        const redeemAmount = weiAmount.mul(scaledToWei);
+        lspContract
+          .redeem(redeemAmount.div(scaledToWei))
+          .then((tx: any) => {
+            return tx.wait(1);
+          })
+          .then(async () => {
+            setAmount("");
+            setLongTokenAmount("");
+            setShortTokenAmount("");
```

Comment:
i think i commented on last pr. should we clear this when we submit transaction rather than once tx is mined?

Reply:
Gotcha, I  can move this before the wait. 

It's a bit weird as it could error out and clear their values, but I think it's not a huge UX issue either way.